copy/copy.go

```diff
@@ -106,9 +113,15 @@ func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageRe
 
 	dest, err := destRef.NewImageDestination(options.DestinationCtx)
 	if err != nil {
-		return errors.Wrapf(err, "Error initializing destination %s", transports.ImageName(destRef))
+		retErr = errors.Wrapf(err, "Error initializing destination %s", transports.ImageName(destRef))
+		return
 	}
-	defer dest.Close()
+	defer func() {
+		if err := dest.Close(); err != nil {
+			retErr = errors.Wrapf(retErr, "dest: %v", err)
```

Comment:
This should only happen if `retErr == nil`; the original value of `retErr` was the primary cause we are failing; that `dest.Close()` has failed may be completely unrelated, or it may be a side effect of the original `retErr` failure.